dds/idl/marshal_generator.cpp

```diff
@@ -2378,29 +2378,25 @@ bool marshal_generator::gen_struct(AST_Structure* node,
     code.push_back("if (!strm.read_delimiter(total_size)) {");
     code.push_back("  return false;");
     code.push_back("}");
+    if (not_final && repr.xcdr2) {
+      be_global->impl_ <<
+        "  size_t start_pos = strm.pos();\n";
+    }
     generate_dheader_code(code, not_final);
```

Comment:
```suggestion
    generate_dheader_code(code, not_final);
    if (not_final) {
      be_global->impl_ <<
        "  size_t start_pos = strm.pos();\n";
    }
```

Since Clayton's PR was merged, we should be getting rid of these `repr` checks since it was decided we can't actually make that optimization in that PR. Also move the stream check after dheader for below.